overrides/default.nix

```diff
@@ -3981,13 +3980,30 @@ lib.composeManyExtensions [
       pydantic = prev.pydantic.overridePythonAttrs
         (old: { buildInputs = old.buildInputs or [ ] ++ [ pkgs.libxcrypt ]; });
 
-      vllm = prev.vllm.overridePythonAttrs (old: {
-        autoPatchelfIgnoreMissingDeps = true;
-
-        # Filter out vllm-nccl-cu12 because vllm-nccl-cu12 will try to download NCCL 2.18.1 from the internet to the ~/.config/vllm/nccl/cu12 directory, which is not allowed in Nix.
+      vllm-nccl-cu12 = prev.vllm-nccl-cu12.overridePythonAttrs (_:
+        # vllm-nccl-cu12 will try to download NCCL 2.18.1 from the internet to
+        # the ~/.config/vllm/nccl/cu12 directory, which is not allowed in Nix,
+        # so we do it ourselves
         # Set VLLM_NCCL_SO_PATH at runtime to specify the nccl so instead.
         # See https://github.com/vllm-project/vllm/issues/4224
-        propagatedBuildInputs = builtins.filter (e: e.pname != "vllm-nccl-cu12") old.propagatedBuildInputs;
+        let
+          soVersion = "2.18.1";
+          soName = "libnccl.so.${soVersion}";
+          libncclSO = pkgs.fetchurl {
+            url = "https://github.com/vllm-project/vllm-nccl/releases/download/v0.1.0/cu12-${soName}";
+            hash = "sha256-AFUDtiuf5Ga2svYLoq0dqkVxGW/8uCo9d6PHdb/NWsg=";
+          };
+        in
+        {
+          preBuild = ''
+            export HOME="$(mktemp -d)"
```

Comment:
To include the SO file, we need
``` suggestion
export HOME="$out"
```

In addition, `vllm` should be patched to locate the SO.

Reply:
What would this change accomplish?

Reply:
I think I see.

The library will end up in a place that would allow `vllm` to be patched to use it.

Reply:
@cpcloud I think excluding `vllm-nccl-cu12` is the easiest approach. When `VLLM_NCCL_SO_PATH ` is unset, `vllm` will use `libnccl.so.2` from rpath as a fallback, according to https://github.com/vllm-project/vllm/blob/e8cc7967ff8a6f8432747a9e87ab451d36e1ff57/vllm/utils.py#L564C42-L564C54 . The default `libnccl.so.2` can be overridden easily in nix. We don't actually need `vllm-nccl-cu12`.

Reply:
Ok, SGTM. Then simply excluding the dependency would work IIUC.